File: pet_hospital_frontend/src/api/bill.ts
import api from './index'

export interface Bill {
    id?: number
    customerId: number
    relatedType: string // PRESCRIPTION
    relatedId: number
    totalAmount: number
    discountAmount: number
    finalAmount: number
    paymentMethod?: string
    status: number // 0:UNPAID, 1:PAID
    createdAt?: string
    paidAt?: string
    customer?: {
        name: string
        level?: number
    }
}

export const billApi = {
    listUnpaid: () =>
        api.get<any, { code: number; data: Bill[] }>('/bills/unpaid'),

    listPaid: () =>
        api.get<any, { code: number; data: Bill[] }>('/bills/paid'),

    createFromPrescription: (id: number) =>
        api.post<any, { code: number; data: null }>(`/bills/create/prescription/${id}`),

    createFromLabTest: (id: number) =>
        api.post<any, { code: number; data: null }>(`/bills/create/lab-test/${id}`),

    createFromSurgery: (id: number) =>
        api.post<any, { code: number; data: null }>(`/bills/create/surgery/${id}`),

    pay: (id: number, paymentMethod: string) =>
        api.post<any, { code: number; data: null }>(`/bills/pay/${id}`, null, {
            params: { paymentMethod }
        })
}

File: pet_hospital_backend/src/main/java/com/pethospital/entity/Bill.java
package com.pethospital.entity;

import com.baomidou.mybatisplus.annotation.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Data
@TableName("bills")
public class Bill {
    @TableId(type = IdType.AUTO)
    private Long id;
    
    private Long customerId;
    
    /**
     * 关联类型: REGISTRATION, PRESCRIPTION
     */
    private String relatedType;
    
    private Long relatedId;
    
    /**
     * 应收金额
     */
    private BigDecimal totalAmount;
    
    /**
     * 优惠金额
     */
    private BigDecimal discountAmount;
    
    /**
     * 实收金额
     */
    private BigDecimal finalAmount;
    
    /**
     * 支付方式: CASH, WECHAT, ALIPAY
     */
    private String paymentMethod;
    
    /**
     * 状态: 0:未付, 1:已付, 2:退款
     */
    private Integer status;
    
    @TableField(fill = FieldFill.INSERT)
    private LocalDateTime createdAt;
    
    private LocalDateTime paidAt;
    
    // 关联字段
    @TableField(exist = false)
    private Customer customer;
}
